starcheck/src/lib/Ska/Starcheck/Obsid.pm

```diff
@@ -2181,52 +2227,26 @@ sub get_agasc_stars {
 #############################################################################################
 
     my $self = shift;
-    my $AGASC_DIR = shift;
+    my $AGASC_FILE = shift;
     my $c;
     return unless ($c = find_command($self, "MP_TARGQUAT"));
-    
-
-    my $agasc_region = Ska::AGASC->new({
-        agasc_dir => $AGASC_DIR,
-        ra => $self->{ra},
-        dec => $self->{dec},
-        radius => 1.3,
-        datetime => $self->{date},
-    });
-
-    my $q_aca = Quat->new($self->{ra}, $self->{dec}, $self->{roll});
 
-    for my $id ($agasc_region->list_ids() )  {
+    # Use Python agasc to fetch the stars into a hash
+    $self->{agasc_hash} = _get_agasc_stars($self->{ra},
+                                           $self->{dec},
+                                           $self->{roll},
+                                           1.3,
+                                           $self->{date},
+                                           $AGASC_FILE);
 
-	my $star = $agasc_region->get_star($id);
-	
-	my ($yag, $zag) = Quat::radec2yagzag(
-					     $star->ra_pmcorrected(), 
-					     $star->dec_pmcorrected(), 
-					     $q_aca);
-	$yag *= $r2a;
-	$zag *= $r2a;
-	if ($star->mag_aca() < -10 or $star->mag_aca_err() < -10) {
-	    push @{$self->{warn}}, sprintf("$alarm Star with bad mag %.1f or magerr %.1f at (yag,zag)=%.1f,%.1f\n",
-					   $star->mag_aca(), $star->mag_aca_err(), $yag, $zag);
+    foreach my $star (values %{$self->{agasc_hash}}) {
+	if ($star->{'mag_aca'} < -10 or $star->{'mag_aca_err'} < -10) {
+	    push @{$self->{warn}}, sprintf(
+                "$alarm Star with bad mag %.1f or magerr %.1f at (yag,zag)=%.1f,%.1f\n",
```

Comment:
Should try to confirm warning still works

Reply:
Looks like this last came up operationally in the OCT1402D products which are too old to run in modern starcheck (and I don't recall if the test here is looking for defects in the agasc that have since been repaired or if there is just the general concern that fields that have stars that have missing data are unreliable).  This won't get tested now.

Reply:
Not testing now is fine.